code/code/misc/combat.cc

```diff
@@ -1333,17 +1333,34 @@ bool TObj::makeScraps()
 
 static void absorb_damage(TBeing *v, wearSlotT part_hit, int *dam)
 {
-  int armor;
-  int abs;
+  TObj *equip = nullptr;
+  TThing *t;
+  int armor = v->acForPos(part_hit) / stats.absorb_damage_divisor[v->isPc() ? PC_STAT : MOB_STAT];
+  int absorb = min(armor/10, *dam);
+
+  if (v->isPc()) {
+    if (!(t = v->equipment[part_hit]) || !(equip = dynamic_cast<TObj *>(t)))
+      absorb = 0;
+    else {
+      if (equip->itemType() == ITEM_WORN) {
+        absorb /= 2;
```

Comment:
You're later on doing float math on absorb, perhaps we want float division here?